src/test/regress/expected/failure_real_time_select.out

```diff
@@ -0,0 +1,198 @@
+-- 
+--  Failure tests for real time select queries
+-- 
+CREATE SCHEMA real_time_select_failure;
+SET search_path TO 'real_time_select_failure';
+SET citus.next_shard_id TO 190000;
+-- Preparation
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SET citus.shard_count to 4;
+SET citus.shard_replication_factor to 1;
+-- create tables
+CREATE TABLE test_table(id int, value_1 int, value_2 int);
+SELECT create_distributed_table('test_table','id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- Populate data to the table
+INSERT INTO test_table VALUES(1,1,1),(1,2,2),(2,1,1),(2,2,2),(3,1,1),(3,2,2);
+-- Kill when the first COPY command arrived, since we have a single placement 
+-- it is expected to error out.
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SET client_min_messages TO ERROR;
+DO $$
+BEGIN
+	SELECT count(*) FROM test_table;
+	
+	EXCEPTION WHEN OTHERS THEN
+		IF SQLERRM LIKE 'failed to execute task%' THEN
+			RAISE 'Task failed to execute';
+		END IF;
+END;
+$$;
+ERROR:  Task failed to execute
+CONTEXT:  PL/pgSQL function inline_code_block line 7 at RAISE
+-- Since the outer query uses the connection opened by the CTE,
+-- killing connection after first successful query should break. 
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY").after(1).kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+DO $$
+BEGIN
+	WITH
+	results AS (SELECT * FROM test_table)
+	SELECT * FROM test_table, results
+	WHERE test_table.id = results.id;
+	EXCEPTION WHEN OTHERS THEN
+		IF SQLERRM LIKE 'failed to execute task%' THEN
+			RAISE 'Task failed to execute';
+		END IF;
+END;
+$$;
+ERROR:  Task failed to execute
+CONTEXT:  PL/pgSQL function inline_code_block line 9 at RAISE
+SET client_min_messages TO DEFAULT;
+-- In parallel execution mode Citus opens separate connections for each shard
+-- so killing the connection after the first copy does not break it.
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY").after(1).kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM test_table;
+ count 
+-------
+     6
+(1 row)
+
+-- Since Citus opens a new connection after a failure within the real time
+-- execution and after(1).kill() kills connection after a successful execution
+-- for each connection, following transaciton does not fail.
+SET citus.multi_shard_modify_mode to sequential;
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY").after(1).kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT citus.clear_network_traffic();
+ clear_network_traffic 
+-----------------------
+ 
+(1 row)
+
+BEGIN;
+SELECT count(*) FROM test_table;
+WARNING:  could not consume data from worker node
+ count 
+-------
+     6
+(1 row)
+
+COMMIT;
+SELECT citus.dump_network_traffic();
```

Comment:
do we need the network dump?  I find those too verbose for adding to the regression test. And, even if you want to add, maybe move to the top of the file as @lithp did in other tests?